articles/ros_node_idl.md

```diff
@@ -97,59 +104,128 @@ This proposal has a number of potential upsides, but it also has some downsides
+The package interface is defined in a separate [XML][xml_wiki] file with suffix `.nodl.xml`. This XML file is exported to the [ament index][ament_index], either manually in the case of python projects or with a helper CMake macro.
+
 The interface may cover only a subset of nodes in a package, as long as the nodes that _are_ covered are done so completely.
 
-Here is an example IDL for a package containing two nodes:
+Here is an example NoDL for a package containing two nodes:
 
-{% include_relative ros2_node_idl/interface_declaration.xml %}
+{% include_relative ros2_nodl/interface_declaration.xml %}
 
-Once an IDL file is written, it is exported from the package manifest:
 
-{% include_relative ros2_node_idl/package.xml %}
+Once an NoDL file is written, it is exported from either `CMakeLists.txt` or `setup.py`.
 
-Note that several IDL files can be exported, allowing for writing an IDL file per node.
+For CMake based builds, the package [`ament_nodl`](https://github.com/ubuntu-robotics/ament_nodl) will provide a a macro `nodl_export_node_description_file()`. That installs the two relevan
 
-### Schema for `package.xml`'s export tag
+{% include_relative ros2_nodl/package.xml %}
 
-#### `interface`
+Note that several NoDL files can be exported, allowing for writing an NoDL file per node.
 
-This is how the package exports its defined IDL for other tools to consume.
+### Exporting a NoDL to the Ament Index
 
-Attributes:
+Per the design philosophy of the [ament index][ament_index], we install files in two locations. In the case of a package named `Foo`, we export our NoDL file `foo.nodl.xml` into the package share directory, `share/foo/foo.nodl.xml`. We then place a marker file, `share/ament_index/nodl_desc/foo`. This is either empty or contains the relative path to the `foo.nodl.xml` file.
 
-**path**:  Path to XML file containing IDL
+#### CMake Macro
 
-Note that the introduction of the `interface` tag within the `export` tag of the package manifest raises a small difficulty with regards to the [REP 149][rep149_export].
-The REP specifies:
+For packages using ament_cmake, the package `ament_nodl` provides the macro `nodl_export_node_description_file` which performs the above for you.
 
-> To avoid potential collisions, an export tag should have the same name as the package which is meant to process it. The content of that tag is up to the package to define and use.
+#### setup.py
 
-Considering the high level abstraction of the IDL, the `interface` tag is not meant for a specific package but rather declares intrinsic properties for anyone to process it.
-However, the keyword is likely solidly descriptive enough to either be accepted as falling under [REP 149][rep149_export] or else to motivate an amendment of the REP.
+In the case of setup.py, the above must be done manually alongside the placement of the package's marker in the ament index.
```

Comment:
There is no equivalent to `nodl_export_node_description_file`? Do we envision one?

Reply:
They don't have anything like that for the package index marker, either. This seems to follow that pattern.